schematic/schemas/json_schema_validation_rule_functions.py

```
@@ -0,0 +1,395 @@
+
+    Returns:
+        A list of ValidationRuleNames
+    """
+    validation_rules = [
+        _VALIDATION_RULES.get(rule_name_string)
+        for rule_name_string in get_names_from_inputted_rules(inputted_rules)
+    ]
+    return [rule.name for rule in validation_rules if rule is not None]
+
+
+def get_names_from_inputted_rules(inputted_rules: list[str]) -> list[str]:
+    """Gets the names from a list of inputted rules
+
+    Arguments:
+        inputted_rules: A list of inputted validation rules from a data model
+
+    Returns:
+        The names of the inputted rules
+    """
+    return [_get_name_from_inputted_rule(rule) for rule in inputted_rules]
+
+
+def _get_parameters_from_inputted_rule(inputted_rule: str) -> Optional[dict[str, str]]:
+    """Creates a dictionary of parameters and values from and input rule string
+
+    Args:
+        input_rule: An inputted validation rule from a data model
+
+    Returns:
+        If the rule exists, a dictionary where
+          the keys are the rule parameters
+          the values are the input rule parameter values
+        Else None
+    """
+    rule_name = _get_name_from_inputted_rule(inputted_rule)
+    rule_values = inputted_rule.split(" ")[1:]
+    rule = _VALIDATION_RULES.get(rule_name)
+    if rule and rule.parameters:
+        return dict(zip(rule.parameters, rule_values))
+    return None
+
+
+def _get_name_from_inputted_rule(inputted_rule: str) -> str:
+    """Gets the name from an inputted rule
+
+    Arguments:
+        inputted_rule: An inputted validation rule from a data model
+
+    Returns:
+        The name of the inputted rule
+    """
+    return inputted_rule.split(" ")[0]
+
+
+def _get_rules_by_names(names: list[str]) -> list[ValidationRule]:
+    """Gets a list of ValidationRules by name of they exist
+
+    Args:
+        names: A list of names of ValidationRules
+
+    Returns:
+        A list of ValidationRules
+    """
+    rules = [_VALIDATION_RULES.get(name) for name in names]
```

hmm I have the same concern here. If a name is not a rule in `_VALIDATION_RULES`, this will also fail silently.